pkg/volume/gce_pd/gce_util.go

```diff
@@ -207,27 +221,29 @@ func createRegionalPD(
 	cloudTags *map[string]string,
 	cloud *gcecloud.GCECloud) error {
 
-	autoZoneSelection := false
-	if replicaZones.Len() != maxRegionalPDZones {
-		replicaZonesList := replicaZones.UnsortedList()
-		if replicaZones.Len() == 1 && replicaZonesList[0] == regionalPDZonesAuto {
-			// User requested automatic zone selection.
-			autoZoneSelection = true
-		} else {
-			return fmt.Errorf(
-				"replica-zones specifies %d zones. It must specify %d zones or the keyword \"auto\" to let Kubernetes select zones.",
-				replicaZones.Len(),
-				maxRegionalPDZones)
-		}
+	zoneCount := replicaZones.Len()
+	if zoneCount > 0 && zoneCount < maxRegionalPDZones {
+		return fmt.Errorf("cannot specify only %d zone(s) for Regional PDs.", zoneCount)
```

Comment:
If `zones` is not present, that should enable `autoZoneSelection` not error.
If `len(zones) > replicationSize`, that should enable `autoZoneSelection`. Which is what you do.
If `len(zones) < replicationSize`, maybe that should also enable `autoZoneSelection`? But I can also be convinced error in this case is ok.

Reply:
if zones is not present, the zoneCount == 0, and it will proceed with auto selection below

Reply:
Ah yes, ok, this LGTM

Reply:
If `zones` is not present, the caller of this function guarantees `replicaZones` is an empty set, and the `zoneCount == 0` case below (equivalent of `autoZoneSelection`) executes.

If `len(zones) < replicationSize`, it's more likely a user error. IMO it's better to notify the user rather than falling back to a behavior a different value already triggers (the `zones == ""` case).

Reply:
If `len(zones) < replicationSize`, it's more likely to be a user error. IMO it's better to notify users of the mistake than falling back to a case triggered by a different parameter value already.